lib/puppet/indirector/yaml.rb

```diff
@@ -19,17 +19,20 @@ def find(request)
   def save(request)
     raise ArgumentError.new("You can only save objects that respond to :name") unless request.instance.respond_to?(:name)
 
-    file = path(request.key)
+    files = path(request.key)
```

Comment:
It's not dead code for other indirected types, but it is uncalled when the subclass of Puppet::Indirector::Yaml is Puppet::Indirector::Report::Yaml.

In order to extend for completeness, we need to figure out the semantics that should be used for each operation:

* save - case introduced here, saves multiple copies of the indirected item, as path returned an array
* find - return the contents of the first element of path? returns the contents of all the elements of path combined in some manner? what happens if one or more of the elements is absent?
* destroy - unlinks all the elements of path? what if one or more is already absent?
* search - searches in order and stops once 'found'?  loads all to combine?